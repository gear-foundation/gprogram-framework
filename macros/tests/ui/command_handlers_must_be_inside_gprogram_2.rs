use gprogram_framework_macros::{command_handlers, gprogram};

#[gprogram]
mod gprogram {
    use super::*;

    mod nested {
        use super::*;

        command_handlers!(
            fn set_value(value: u32) -> Result<(), ()> {
                Ok(())
            }
        );
    }
}

fn main() {}
