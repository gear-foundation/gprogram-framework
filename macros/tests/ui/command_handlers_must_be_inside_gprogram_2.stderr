error: The `command_handlers` block must be on the top level of `gprogram` module
  --> tests/ui/command_handlers_must_be_inside_gprogram_2.rs:11:13
   |
11 | /             fn set_value(value: u32) -> Result<(), ()> {
12 | |                 Ok(())
13 | |             }
   | |_____________^
